api/src/main/java/org/openmrs/validator/PersonValidator.java

```diff
@@ -107,7 +108,36 @@ public void validate(Object target, Errors errors) {
 		
 		ValidateUtil.validateFieldLengths(errors, Person.class, "gender", "personVoidReason");
 	}
-	
+
+	/**
+	 * Checks if the death date is in accordance with dead status.
+	 *
+	 * @param errors Stores information about errors encountered during validation
+	 * @param deathDate the deathdate to validate
+	 * @param dead is the dead status
+	 */
+	private void validateDeathDateDead(Errors errors, Date deathDate, boolean dead) {
+		if(deathDate == null || dead == true) {
```

Comment:
@dkayiwa  there shouldn't be a problem , bcoz that way the person is not dead and no deathdate is being associated with it.
I noticed a thing in a lot of test cases that the deathdate has been initialized with new Date(), which return an instance of current date, which means that the deathdate is not null. also in these tests dead is not set to true. i think this may be a problem.